### PR TITLE
[ADF-4807] Automation tests for date min/max validation

### PR DESCRIPTION
**Pull request by marouanbentaleb**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:
Automation tests for date widget min/max validation

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-4775

### CODE REVIEW

File: e2e/pages/adf/demo-shell/process-services/formDemoPage.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        await BrowserActions.click(this.formCloudEditor);
+    }
+
+    async goToRenderedForm(): Promise<void> {
+        await BrowserActions.click(this.formCloudRender);
+    }
+
+    async setConfigToEditor(text): Promise<void> {
+        await this.goToEditor();
+        await browser.sleep(2000);
```
Comment (eromano) [inline e2e/pages/adf/demo-shell/process-services/formDemoPage.ts]: maybe you can change this sleep with checking if some elements of that page are present

File: scripts/check-env/check-activiti-env.js [hunk 2]
```diff
@@ -24,8 +24,8 @@ async function main() {
         .option('--host [type]', 'Remote environment host adf.lab.com ')
         .option('--oauth [type]', 'oauth host')
         .option('--client [type]', 'clientId ')
-        .option('-p, --password [type]', 'password ')
```
Comment (eromano) [inline scripts/check-env/check-activiti-env.js]: why is this modify needed?

File: scripts/travis/e2e/process-services-cloud-e2e.sh [hunk 3]
```diff
@@ -11,17 +11,17 @@ CONTEXT_ENV="process-services-cloud"
 AFFECTED_LIBS="$(./scripts/affected-libs.sh -gnu -b $TRAVIS_BRANCH)";
 AFFECTED_E2E="$(./scripts/git-util/affected-folder.sh -b $TRAVIS_BRANCH -f "e2e/$CONTEXT_ENV")";
 
-RUN_E2E=$(echo ./scripts/test-e2e-lib.sh -host http://localhost:4200 -proxy "$E2E_HOST_BPM" -u "$E2E_USERNAME" -p "$E2E_PASSWORD" -e "$E2E_EMAIL" -host_sso "$E2E_HOST_SSO" -host_bpm "$E2E_HOST_BPM" -host_identity "$E2E_HOST_IDENTITY" -identity_admin_email "$E2E_ADMIN_EMAIL_IDENTITY" -identity_admin_password "$E2E_ADMIN_PASSWORD_IDENTITY" --use-dist -gnu )
```
Comment (eromano) [inline scripts/travis/e2e/process-services-cloud-e2e.sh]: why is this modify needed?

Comment (marouanbentaleb) [reply scripts/travis/e2e/process-services-cloud-e2e.sh]: Until now we've been using admin.adf to login to process-services-cloud e2e, which is not a good practice. I'm trying to change it to hruser.

File: e2e/process-services-cloud/task-form-cloud-component.e2e.ts [hunk 4]
```diff
@@ -168,60 +161,64 @@ describe('Task form cloud component', () => {
...
+            await tasksCloudDemoPage.taskListCloudComponent().checkContentIsDisplayedByName(formValidationsTask.entry.name);
+            await tasksCloudDemoPage.taskListCloudComponent().selectRow(formValidationsTask.entry.name);
+            await taskFormCloudComponent.checkFormIsDisplayed();
+            await taskFormCloudComponent.formFields().checkFormIsDisplayed();
+            await taskFormCloudComponent.formFields().checkWidgetIsVisible('Text0tma8h');
+            await taskFormCloudComponent.formFields().checkWidgetIsVisible('Date0m1moq');
+            await taskFormCloudComponent.formFields().checkWidgetIsVisible('Number0klykr');
+            await taskFormCloudComponent.formFields().checkWidgetIsVisible('Amount0mtp1h');
+
+            await expect(await (await taskFormCloudComponent.getCompleteButton()).isEnabled()).toBe(false);
```
Comment (adinapitul) [inline e2e/process-services-cloud/task-form-cloud-component.e2e.ts]: Maybe create a method like isCompleteButtonEnabled()

Comment (gmandakini) [reply e2e/process-services-cloud/task-form-cloud-component.e2e.ts]: do we need 3 awaits here?

File: scripts/travis/e2e/process-services-cloud-e2e.sh [hunk 5]
```diff
@@ -11,7 +11,7 @@ CONTEXT_ENV="process-services-cloud"
 AFFECTED_LIBS="$(./scripts/affected-libs.sh -gnu -b $TRAVIS_BRANCH)";
```
Comment (eromano) [inline scripts/travis/e2e/process-services-cloud-e2e.sh]: can we moev this modify in a separate PR?

Comment (eromano) [reply scripts/travis/e2e/process-services-cloud-e2e.sh]: @marouanbentaleb please remove from here

**Pull request merged**